src/main/java/com/fasterxml/jackson/databind/deser/std/UntypedObjectDeserializer.java

```diff
@@ -196,7 +195,7 @@ public void resolve(DeserializationContext ctxt) throws JsonMappingException
         if ((_stringDeserializer == null) && (_numberDeserializer == null)
                 && (_mapDeserializer == null) && (_listDeserializer == null)
                 &&  getClass() == UntypedObjectDeserializer.class) {
-            return Vanilla.instance(preventMerge);
+            //return Vanilla.instance(preventMerge);
```

Comment:
Alas, Vanilla instance is performance-wise significantly faster than having to consider possible custom deserializers... so ultimately I'd like to retain this distinction. And/or at least measure performance difference in case I am wrong that there is significant difference.